### PR TITLE
Add support for origin-access-control

### LINKED ISSUE #244: Add support for origin-access-control to replace origin-access-identity
Have a question? Please checkout our [Slack Community](https://slack.cloudposse.com) or visit our [Slack Archive](https://archive.sweetops.com/). 

[Slack Community](https://slack.cloudposse.com)

## Describe the Feature

Add the capability of enabling origin access control as described in https://aws.amazon.com/blogs/networking-and-content-delivery/amazon-cloudfront-introduces-origin-access-control-oac/, https://aws.amazon.com/about-aws/whats-new/2022/08/amazon-cloudfront-origin-access-control/, https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-restricting-access-to-s3.html and https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_origin_access_control.

## Expected Behavior

Origin access control is used between Cloudfront and the S3 origin instead of origin access identity.  This 

## Use Case

Allows the use of SSE aws:kms and customer-managed KMS keys that can be shared across accounts that, in turn, allows for cross-account deployments with KMS-encrypted S3 buckets.

## Describe Ideal Solution

The S3 bucket that is configured as S3 origin has the proper policy assigned to allow Cloudfront to access it via OAC instead of OAI.

## Alternatives Considered

None.

## Additional Context

Once I used your module to create a Cloudfront distribution with OAI, I then followed the instructions in the Cloudfront developer guide to  update the distribution with the OAC I created using Terraform cloudfront_origin_access_control.  
cloudfront get-distribution-config --id dist-id --output yaml > dist-config.yaml
Edit as specified in https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-restricting-access-to-s3.html section labeled *To attach an OAC to an S3 bucket origin in an existing distribution (CLI with input file)*
cloudfront update-distribution --id dist-id --cli-input-yaml file://dist-config.yaml

### PR DESCRIPTION
**Pull request by rankin-tr**
## what

- add Origin Access Control feature
  - add `var.origin_access_type` to enable Origin Access Identity or Origina Access Control policy
  - add `aws_cloudfront_origin_access_control.default` resource
  - add `origin_access_control_id` argument to origin config on `aws_cloudfront_distribution.default`
- update example code
- update README

## why

- provide the ability to make use of an Origin Access Control
  - retain default origin access identity behavior
- AWS recommends using origin access control
- Origin Access Identities are flagged in [AWS Security Hub](https://docs.aws.amazon.com/securityhub/latest/userguide/cloudfront-controls.html#cloudfront-13)

## references

- Closes #244 
- https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-restricting-access-to-s3.html

### CODE REVIEW

**Comment by philchristensen** [comment]
Just adding a positive review; I tried this branch myself in a project and it worked great. Thanks @rankin-tr !

File: main.tf [hunk 1]
```diff
@@ -419,8 +469,7 @@ resource "aws_cloudfront_distribution" "default" {
 
   depends_on = [
     aws_s3_bucket.origin,
-    aws_s3_bucket_ownership_controls.origin,
-    time_sleep.wait_for_aws_s3_bucket_settings
```
Comment (dudymas) [inline main.tf:423]: Is there a reason to drop the time_sleep? Even if OAC doesn't require this, others might still be using OAI

Comment (rankin-tr) [reply main.tf:423]: It was necessary to drop `time_sleep` because it created circular dependencies

**Review by dudymas — DISMISSED** [review]

**Comment by GabisCampana** [comment]
@Nuru @dudymas

**Review by dudymas — APPROVED** [review]

**Review by Benbentwo — APPROVED** [review]
LGTM Thanks!

**Pull request merged**